### PR TITLE
test: make the symbols-leak-test executable

### PR DESCRIPTION
**Pull request by whot**
Python leaks like crazy when run under valgrind. But if we make the script
executable **and** it has uses the env invocation (i.e. #!/usr/bin/env python3),
the leaks disappear. This is not the case for a shebang of /usr/bin/python3.

Why exactly this is the case I'm not sure but executables we plan to run
should have the exec bit set. So this is a janitor patch with the nice side
effect of fixing our valgrind runs.

### CODE REVIEW

**Comment by whot** [comment]
First patch is the immediate fix, second patch is is the "proper" fix, at least if we're happy to have our own defined valgrind setup. Feel free to cherry-pick or squash. Unfortunately meson doesn't have the ability to exclude a test from a test setup, so we can't avoid the externally-provided `--no-suite xyz` argument.

File: meson.build [hunk 1]
```diff
@@ -501,6 +502,20 @@ if get_option('enable-xkbregistry')
...
+if valgrind.found()
+    add_test_setup('valgrind',
+        exe_wrapper: [valgrind,
+                       '--leak-check=full',
+                       '--track-origins=yes',
+                       '--gen-suppressions=all',
+                       '--error-exitcode=99'],
+        timeout_multiplier : 10)
+else
+    message('valgrind not found, disabling valgrind test suite')
```
Comment (bluetech) [inline meson.build:515]:
```suggestion
    message('valgrind not found, disabling valgrind test setup')
```

**Comment by bluetech** [comment]
Merged (with the comment fixed), thanks!

**Comment by whot** [comment]
huh, that explains it. thanks for finding that, I had wondered about it.

**Pull request closed without merging**